lib/types/array.js

```diff
@@ -33,14 +33,41 @@ function MongooseArray(values, path, doc) {
   var keysMA = Object.keys(MongooseArray.mixin);
   var numKeys = keysMA.length;
   for (var i = 0; i < numKeys; ++i) {
-    arr[keysMA[i]] = MongooseArray.mixin[keysMA[i]];
+    Object.defineProperty(arr, keysMA[i], {
```

Comment:
this will significantly decrease performance

Reply:
Yeah that's why we're not merging this. No transactional `defineProperty()` allowed :)